Source/santa-driver/SantaCache.h

```diff
@@ -186,6 +185,35 @@ template<typename KeyT, typename ValueT> class SantaCache {
     return count_;
   }
 
+  /**
+    Fill in the per_bucket_counts array with the number of entries in each bucket.
+
+    The per_buckets_count array will contain the per-bucket counts, up to the number
+    in array_size. The start_bucket parameter will determine which bucket to start off
+    with and upon return will contain either 0 if no buckets are remaining or the next
+    bucket to begin with when called again.
+  */
+  void bucket_counts(uint16_t *per_bucket_counts, uint16_t *array_size, uint64_t *start_bucket) {
+    uint64_t start = *start_bucket;
+    if (*start_bucket + *array_size > bucket_count_) *array_size = bucket_count_;
+
+    for (uint16_t i = 0; i < *array_size; ++i) {
+      int16_t count = 0;
+      struct bucket *bucket = &buckets_[start + i];
+      lock(bucket);
+      struct entry *entry = (struct entry *)((uintptr_t)bucket->head - 1);
+      while (entry != nullptr) {
+        if (entry->key != 0) count++;
+        entry = entry->next;
+      }
+      unlock(bucket);
+      (*start_bucket)++;
```

Comment:
nit: To save a couple start_bucket dereferences... 
Increment `start` on l202. You could get rid of this line.
On L214: if (start >= bucket_count_) *start_bucket = 0;

Reply:
Done